clone.js

```diff
@@ -7,7 +7,7 @@ function clone (obj) {
     return obj
 
   if (obj instanceof Object)
-    var copy = { __proto__: obj.__proto__ }
+    var copy = { __proto__: Object.getPrototypeOf(obj) }
```

Comment:
@ljharb No concern about `Object.getPrototypeOf` not existing?

Reply:
I'm concerned about that as well but figured it was implied by my other comment.

Reply:
The special-case handling code for `fs.read` is guarded with a check and won't crash, and this is differrent.

I could surely add a check here, but which engines do you have in mind? Afaik all Node.js versions down to 0.8 have `Object.getPrototypeOf` implemented.

Reply:
Yeah, fair enough - given that even node 0.4 supports it, and this isn't a browser lib, it's not a concern.

Reply:
I added a polyfill for `Object.getPrototypeOf`, so I think this can be resolved.

Reply:
@ljharb that said, even Babel is broken on environments with no `Object.getPrototypeOf`.